shepherd/query_expansion/bte/expansion.py

```diff
@@ -33,69 +69,84 @@ def get_params(query_graph) -> tuple[str, str, str, str, str, dict[str, str]]:
+    if subject_type is not None:
+        subject_types.add(subject_type.removeprefix("biolink:"))
+    if object_type is not None:
+        object_types.add(object_type.removeprefix("biolink:"))
+    if predicate is not None:
+        predicates.add(predicate.removeprefix("biolink:"))
 
     with open(Path(__file__).parent / "templateGroups.json", "r") as file:
-        templateGroups = json.load(file)
+        templateGroups = parse_obj_as(list[TemplateGroup], json.load(file))
 
     template_paths = {
         path.name: path
         for path in (Path(__file__).parent / "templates").rglob("*.json")
     }
 
-    # TODO: strict typing
-    matched_paths = set()
+    matched_paths: set[Path] = set()
     for group in templateGroups:
-        conditions = []
-        conditions.append(len(subject_types.intersection(group["subject"])) > 0)
-        conditions.append(len(object_types.intersection(group["object"])) > 0)
-        conditions.append(len(predicates.intersection(group["predicate"])) > 0)
+        conditions: list[bool] = []
+        conditions.append(len(subject_types.intersection(group.subject)) > 0)
+        conditions.append(len(object_types.intersection(group.object)) > 0)
+        conditions.append(len(predicates.intersection(group.predicate)) > 0)
         conditions.append(  # Qualifiers (if they exist) are satisfied
             all(
-                group.get("qualifiers", {}).get(qualifier_type, False) == value
+                (group.qualifiers or {}).get(qualifier_type, False) == value
                 for qualifier_type, value in qualifiers.items()
             )
         )
 
         if all(conditions):
-            for template in group["templates"]:
+            for template in group.templates:
                 matched_paths.add(template_paths[template])
 
     return list(matched_paths)
 
 
 def fill_templates(
-    paths: list[Path], subject_curie, object_curie
-) -> list[dict[str, Any]]:
-    filled_templates = []
+    paths: list[Path], subject_curie: Optional[CURIE], object_curie: Optional[CURIE]
+) -> list[Query]:
+    filled_templates: list[Query] = []
     for path in paths:
         with open(path, "r") as file:
-            template = json.load(file)
+            template = Query.parse_obj(json.load(file))
         if subject_curie is not None:
-            template["message"]["query_graph"]["nodes"]["creativeQuerySubject"][
-                "ids"
-            ] = [subject_curie]
+            cast(QueryGraph, template.message.query_graph).nodes[
+                "creativeQuerySubject"
+            ].ids = HashableSequence(__root__=[subject_curie])
         if object_curie is not None:
-            template["message"]["query_graph"]["nodes"]["creativeQueryObject"][
-                "ids"
-            ] = [object_curie]
+            cast(QueryGraph, template.message.query_graph).nodes[
+                "creativeQueryObject"
+            ].ids = HashableSequence(__root__=[object_curie])
```

Comment:
Is this the correct way to create the ids value when working in reasoner-pydantic? It seems to test fine.